ci/jenkins/test-rancher.sh

```diff
@@ -0,0 +1,298 @@
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+while [[ $# -gt 0 ]]
+do
+key="$1"
+
+case $key in
+    --cluster-name)
+    CLUSTER_NAME="$2"
+    shift 2
+    ;;
+    --workdir)
+    WORKDIR="$2"
+    shift 2
+    ;;
+    --testcase)
+    TESTCASE="$2"
+    shift 2
+    ;;
+    --registry)
+    DOCKER_REGISTRY="$2"
+    shift 2
+    ;;
+    -h|--help)
+    print_usage
+    exit 0
+    ;;
+    *)    # unknown option
+    echoerr "Unknown option $1"
+    exit 1
+    ;;
+esac
+done
+
+if [[ "$WORKDIR" != "$DEFAULT_WORKDIR" && "$KUBECONFIG_PATH" == "$DEFAULT_KUBECONFIG_PATH" ]]; then
+    KUBECONFIG_PATH=${WORKDIR}/.kube/config
+fi
+NO_PULL=
+if [[ "$DOCKER_REGISTRY" != "" ]]; then
+    # Image pulling policy of Sonobuoy is 'Always' by default. With dockerhub rate limit, sometimes it is better to use
+    # cache when registry is used.
+    IMAGE_PULL_POLICY="IfNotPresent"
+    # If DOCKER_REGISTRY is non null, we ensure that "make" commands never pull from docker.io.
+    NO_PULL=1
+fi
+export NO_PULL
+
+E2ETEST_PATH=${WORKDIR}/kubernetes/_output/dockerized/bin/linux/amd64/e2e.test
+RANCHER='sudo /home/ubuntu/rancher-v2.7.0/./rancher'
+
+function rancher_login {
+    ${RANCHER} login $RANCHER_URL --token $RANCHER_ACCESS_TOKEN --skip-verify --context $RANCHER_UPSTREAM_CONTEXT
+}
+
+function fetch_kubeconfig {
+    ${RANCHER} cluster kf $CLUSTER_NAME -o yaml > "$1"
+}
+
+function export_govc_env_var {
+    export GOVC_URL=$GOVC_URL_FOR_RANCHER
+    export GOVC_USERNAME=$GOVC_USERNAME
+    export GOVC_PASSWORD=$GOVC_PASSWORD
+    export GOVC_INSECURE=1
+    export GOVC_DATACENTER=$GOVC_DATACENTER
+    export GOVC_DATASTORE=$GOVC_DATASTORE
+}
+
+function clean_vm_agent {
+    echo "====== Cleanup Antrea-Agent Installation on the VM ======"
+    echo "Revert to snapshot defaultState"
+    export_govc_env_var
+    ${RANCHER} nodes | awk 'NR>1 {print $2}' | while read HOSTNAME; do
+        govc snapshot.revert -k=true -vm=${HOSTNAME} defaultState
+    done
+}
+
+function deliver_antrea {
+    echo "====== Cleanup Antrea Installation ======"
+    clean_vm_agent
+
+    echo "====== Fetch and Export Kubeconfig ======"
+    KUBECONFIG_PATH="$(readlink -f .)/kubeconfig"
+    fetch_kubeconfig $KUBECONFIG_PATH
+    export KUBECONFIG=$KUBECONFIG_PATH
+
+    echo "====== Building Antrea for the Following Commit ======"
+    export GO111MODULE=on
+    export GOPATH=${WORKDIR}/go
+    export GOROOT=/usr/local/go
+    export GOCACHE="${WORKSPACE}/../gocache"
+    export PATH=${GOROOT}/bin:$PATH
+
+    git show --numstat
+    make clean
+    ${CLEAN_STALE_IMAGES}
+    if [[ ! "${TESTCASE}" =~ "e2e" && "${DOCKER_REGISTRY}" != "" ]]; then
+        docker pull "${DOCKER_REGISTRY}/antrea/sonobuoy-systemd-logs:v0.3"
+        docker tag "${DOCKER_REGISTRY}/antrea/sonobuoy-systemd-logs:v0.3" "sonobuoy/systemd-logs:v0.3"
```

Comment:
Do we need to run this on rancher-agent node? I think maybe it's needed only on cp nodes and worker nodes.